writers/postgres/init.go

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (c) 2018
+// Mainflux
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package postgres
+
+import (
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq" // required for SQL access
+	migrate "github.com/rubenv/sql-migrate"
+)
+
+// Config defines the options that are used when connecting to a PostgreSQL instance
+type Config struct {
+	Host        string
+	Port        string
+	User        string
+	Pass        string
+	Name        string
+	SSLMode     string
+	SSLCert     string
+	SSLKey      string
+	SSLRootCert string
+}
+
+// Connect creates a connection to the PostgreSQL instance and applies any
+// unapplied database migrations. A non-nil error is returned to indicate
+// failure.
+func Connect(cfg Config) (*sqlx.DB, error) {
+	url := fmt.Sprintf("host=%s port=%s user=%s dbname=%s password=%s sslmode=%s sslcert=%s sslkey=%s sslrootcert=%s", cfg.Host, cfg.Port, cfg.User, cfg.Name, cfg.Pass, cfg.SSLMode, cfg.SSLCert, cfg.SSLKey, cfg.SSLRootCert)
+
+	db, err := sqlx.Open("postgres", url)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := migrateDB(db); err != nil {
+		return nil, err
+	}
+
+	return db, nil
+}
+
+func migrateDB(db *sqlx.DB) error {
+	migrations := &migrate.MemoryMigrationSource{
+		Migrations: []*migrate.Migration{
+			{
+				Id: "messages_1",
+				Up: []string{
+					`CREATE TABLE IF NOT EXISTS messages (
+						channel       UUID,
+						subtopic      VARCHAR(254),
+						publisher     UUID,
+            protocol      TEXT,
+						name          TEXT,
+            unit          TEXT,
+            value         FLOAT,
+            string_value  TEXT,
+            bool_value    BOOL,
+            data_value    TEXT,
+            value_sum     FLOAT,
+            time          FlOAT,
+            update_time   FLOAT,
+            link          TEXT,
+						PRIMARY KEY (channel)
+					)`,
```

Comment:
Please fix schema formatting. More importantly, `channel` cannot be message primary ID (it's not unique).